components/chapters/SignUpModal.tsx

```diff
@@ -0,0 +1,132 @@
+'use client'
+
+import Image from 'next/image'
+import { useState } from 'react'
+import Modal from 'react-modal'
+const crypto = require('crypto')
+const Secp256k1 = require('@lionello/secp256k1-js')
+
+export const SignUpModal = (props) => {
+  function generateKeyPair() {
+    const privateKeyBuf = crypto.randomBytes(32)
+    const privateKey = Secp256k1.uint256(privateKeyBuf, 16)
+
+    const publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
+
+    return {
+      privateKey,
+      publicKey
+    };
+  }
+
+  const customStyles = {
+    content: {
+      top: '50%',
+      left: '50%',
+      right: 'auto',
+      bottom: 'auto',
+      marginRight: '-50%',
+      transform: 'translate(-50%, -50%)',
+      backgroundColor: '#32455D',
+      color: '#fff',
+      width: 550,
+    },
+  }
+
+  const [keyPair, _] = useState(generateKeyPair());
+
+  let [copied, setCopied] = useState(false)
+  let [avatar, setAvatar] = useState(1)
+
+  function copy() {
+    navigator.clipboard.writeText(keyPair.privateKey.toString(16))
+    setCopied(true)
+
+    setTimeout(() => {
+      setCopied(false)
+    }, 2000)
+  }
+
+  return (
+    <Modal
+      isOpen={props.open}
+      style={customStyles}
+      contentLabel="Example Modal">
+
+        <div className="float-right flex justify-end">
+          <button onClick={props.onClose}>
```

Comment:
Eventually we want proper aria attributes for buttons to make the UI accessible. For now, a title tag should be enough to explain what this button does..